sunpy/time/tests/test_unixtai58.py

```diff
@@ -0,0 +1,50 @@
+import pytest
+
+from astropy.time import Time, TimeUnixTai
+from astropy.time.formats import erfa
+
+# This registers the TimeTAICDS format with astropy
+from sunpy.time import TimeUnixTai58  # NOQA
+
+
+def test_time_t0():
+    """
+    Test that the offset at the epoch is zero. Note that the offset
+    will be zero in both TAI and UTC
+    """
+    t = Time('1958-01-01 00:00:00', format='iso', scale='tai')
+    assert t.tai.unix_tai_58 == 0.0
+    with pytest.warns(erfa.ErfaWarning, match='dubious year'):
+        assert t.utc.unix_tai_58 == 0.0
+
+
+def test_tai_utc_offset():
+    """
+    Test that the offset between TAI and UTC is 10 s in 1972-01-01, when UTC
+    was defined
+    """
+    t1 = Time('1972-01-01 00:00:00', scale='tai', format='iso')
+    t2 = Time('1972-01-01 00:00:00', scale='utc', format='iso')
+    assert t2.unix_tai_58 - t1.unix_tai_58 == 10.0
+
+
+@pytest.mark.parametrize('time', [
```

Comment:
This would be a good place to use `hypothesis` to generate test data. Not a blocking suggestion though.